bluecellulab/circuit_simulation.py

```diff
@@ -116,6 +117,8 @@ def __init__(
         self.spike_threshold = self.circuit_access.config.spike_threshold
         self.spike_location = self.circuit_access.config.spike_location
 
+        self.projections: Optional[List[str]] = None
```

Comment:
With python 3.9+ `list` can be used as a type  https://peps.python.org/pep-0585/

E.g. this can be `Optional[list[str]] = None` or `list[str] | None = None`

You may need to add `from __future__ import annotations` for python 3.8